### ISSUE TITLE
Setting Windows for use with WiFi ESP8266 connection

### ISSUE THREAD

**Issue by Seeelefant** [mavlink/qgroundcontrol#7189]
The following template is for Issue/Bug reporting only. https://docs.qgroundcontrol.com/en/Support/Support.html#github-issues
I am waiting forever for the Wifi connection:
![image](https://user-images.githubusercontent.com/30323691/51714753-8f89b400-2036-11e9-86c0-db55ea123851.png)
from a Windows 10 box.

The hardware is working, as shown here:
![image](https://user-images.githubusercontent.com/30323691/51714930-1179dd00-2037-11e9-95bd-cff40e89b059.png)

With an Android tablet I a,m able to connect.

Please advice
--Seeelefant

For questions about how to use or build QGC see: http://qgroundcontrol.com/#resources

## Expected Behavior
Please tell us what should happen as detailed as possible.

## Current Behavior
Please describe what happens instead of the expected behavior.

## Steps to Reproduce:
Please provide an unambiguous set of steps to reproduce the current behavior

## System Information
When posting bug reports, include the following information

* Operating System: Windows 10
* QGC Version: 3.4.4
* QGC build: stable]
* Flight Controller: Pixhawk 2
* Autopilot (with version): [e.g., PX4 1.8.12 ]

## Detailed Description
Provide further details about your issue/bug.

## Log Files and Screenshots

* [QGC Console Logs](https://docs.qgroundcontrol.com/en/Support/Support.html#reporting-bugs)
* Autopilot logs when available (post a link).
* Screenshots of QGC to help identify the current issue/bug behavior.

**Comment by Polarbeargo** [comment]
Hello @Seeelefant ,
I had bumped into the same question as you do with QgroundControl v 3.4.4 on Windows 10. I just uninstall the QgroundControl v3.4.4 and reinstall it again turn off the firewall then it works well include rtsp video streaming.

**Comment by Seeelefant** [comment]
Hi Polarbeargo,
thank you, now it is working. It was a problem with the windows firewall, I enabled QGroundcontrol within the Windows Defender Firewall.
Best
--Seeelefant

**Comment by minerran1** [comment]
I'm having the same (I think...) issue.  Can someone please explain how this has to be setup to work?  I see no wifi setup anywhere in QGroundControl.   I can connect via usb to my pixracer.  With the wifi module plugged into the pixracer and usb disconnected (pixracer powered by battery), I expected QGroundControl to find the wifi and connect.  I'm using windows 7 and in network and sharing center, I see the "pixracer"  network.  I can connect to it on the PC by entering the default password.  So far so good but QGroundControl still does not connnect.  Not sure what Seeelefant means by "enabling QGroundControl "within" windows defender.  I opened windows defender and I see no way to "enable" QGroundControl, whatever that means.  Can somebody please explain?

**Comment by Seeelefant** [comment]
Hi at least for Windows 10 the settings look like this:
![image](https://user-images.githubusercontent.com/30323691/53453363-283f9500-3a24-11e9-9d2e-87d70ff260a9.png)

**Comment by minerran1** [comment]
Thanks for replying.  My German from college is a little "rusty" but I get the idea.  I will search to see how to do this for my version.   Danke Schon!

**Comment by DonLakeFlyer** [comment]
@hamishwillee FYI: This is the first I'm hearing of some Windows Defender thing blocking QGC WiFi connections from working right out of the box. Maybe something for the docs?

**Comment by minerran1** [comment]
It didn't fix my problem, investigating.....

**Comment by dogmaphobic** [comment]
For what's worth, I was never able to get WiFi/UDP working on Windows. I can see the packets being sent but they never reach the application (QGC). I tried turning off the firewall (which is burried under an ungodly amount of settings pages) but nothing ever worked.

**Comment by hamishwillee** [comment]
Docs for this in https://github.com/mavlink/qgc-user-guide/pull/199
It is more generic than Windows Defender - I use Norton and I can see that the app has been granted access.

**Comment by DonLakeFlyer** [comment]
> Would have been nice to have had the ability to actually choose the connection (ie. wifi or usb) instead of the only choice being an automatic connection.

You can turn off any flavor of autoconnect from Settings/General page. You can set up manual links from Settings/Comm Links. The point is that if you are using a standard connection type there is no need to do anything it just works. Creating a manual link isn't going to make it work either if auto doesn't work. Manual links are almost never needed. Only if say you WiFi setup uses ports which differ from the standard or say the radio you are connecting to runs at a baud rate different from the default.

**Comment by DonLakeFlyer** [comment]
I'm reopening to try WiFi on Windows to see if there is some additionals sequence of steps of futzing with the OS to get things to work. Then we can doc that if so. That said, this is the first I'm hearing of WiFi problems on Windows.

**Comment by DonLakeFlyer** [comment]
> It is more generic than Windows Defender - I use Norton and I can see that the app has been granted access.

Ah. So you have WiFi on windows working then right. I'll close this then.

**Comment by hamishwillee** [comment]
> Ah. So you have WiFi on windows working then right. I'll close this then.

No I don't - I don't have a wifi telemetry module (just RC one). What I was saying is that I can see in Norton where the app is granted firewall access. I can't see it in windows defender because norton overrides those settings.

It sounds from what @dogmaphobic said that Wifi can be a problem on Windows outside the scope of this issue.

**Comment by Seeelefant** [comment]
Just a additional remark: I have a WiFi stick attached to my PC, with the new firewall settings QGroundcontrol  gets contact to my drone without any problem. Before I made these changes I had connection problem on all windows PCs whereas Mission Planner from the Arducopters worked out of the box.

**Comment by minerran1** [comment]
> 
> 
> > Would have been nice to have had the ability to actually choose the connection (ie. wifi or usb) instead of the only choice being an automatic connection.
> 
> You can turn off any flavor of autoconnect from Settings/General page. You can set up manual links from Settings/Comm Links. The point is that if you are using a standard connection type there is no need to do anything it just works. Creating a manual link isn't going to make it work either if auto doesn't work. Manual links are almost never needed. Only if say you WiFi setup uses ports which differ from the standard or say the radio you are connecting to runs at a baud rate different from the default.

Don,  I believe you but I spent an hour painstakingly looking thru parameter settings, trying to find something obvious and could not.  I'm guessing that you're a developer on this very familiar with the c code inside and out, knowing what every parameter does.   I don't have that experience and rely on the manual or what is intuitively presented in the settings.  Another thing,  its not possible to change anything without first having a vehicle connection based on what I'm seeing.   So I cant even get access to the parameter settings without first connecting via usb.  I concluded that "MAVlink" was related to wifi so I spent a lot of time there, looking for a setting.   Thanks for your opinion and comment.   I'm giving up on wifi.  Not worth the effort of spending endless hours of guessing or trial + error.

**Comment by DonLakeFlyer** [comment]
@hamishwillee Might want to have a discussion with @minerran1 as to what could be made better in the docs to make this less confusing.

There are two types of settings:
* QGC Application Settings - These only relate to how QGC works, not how the vehicle works. They are available from the Settings page.
* Vehicle Settings - These relate to how the Vehicle works. They are actually stored on the vehicle (not in QGC) so you can't do anything with them unless you are connected to a vehicle. They are available from the Setup page.

The important point is that if you are using a standard WiFi thingy like an ESP8266 which has the standard mavlink firmware on it other than getting the connection setup in the OS there is nothing else you need to do. There is nothing else in QGC to look for or change. There is nothing that requires a connection that needs changing. It should just work. If it doesn't there is a hardware or OS problem somewhere as opposed to a ground station problem. Any debugging of that sort of thing needs to happen at the OS level.

**Comment by hamishwillee** [comment]
@DonLakeFlyer Sure. 

1. For this particular case we do currently [say in the quickstart](https://docs.qgroundcontrol.com/en/getting_started/quick_start.html): *Attach your vehicle to the ground station device via USB, through a telemetry radio, or over WiFi. QGroundControl should detect your vehicle and connect to it automatically.*

Which I guess covers the main cases except "what will happen if you have wifi and USB connection"?

What could we have done differently? I think probably adding an explicit connection doc here:
![image](https://user-images.githubusercontent.com/5368500/53610644-b2a01880-3c1f-11e9-8d7e-ae3035b9869e.png)
At least then people would have an obvious place to look for connection information, and once finding that would be able to start looking for host computer issues rather than QGC issues.
We could also add the troubleshooting note to the quick start too. 

@minerran1 Do you think that would have helped/would help?

2. Getting to grips with QGC

@minerran1 Did you just jump into QGC .... or "have you been guided by the docs". 
I am by no means advocating that the docs are great but if you haven't used them then talking about how they could be better is probably not going to help.

If you have been using them then would be great to gather your thoughts. Would slack or email or some other system suit you best?

**Comment by minerran1** [comment]
First let me answer your questions.   I always try to use new applications without first reading documentation to see if I can figure it out.  Sometimes its intuitive.   If I run into issues, I then look for documentation which I found.  I did find the instructions you posted above.  I was not 100% sure what "ground station device" was but it has to be a computer or mobile phone with QGroundControl installed.  I would have wrote it that way explicitly but thats just me.   I favor verbose whereas most people these days favor the opposite.   As far as commenting on how to "write it better/help", I cannot offer suggestions until I know why my wifi isn't working.

Now you said,   "It should just work."
1. Does "should just work" mean that after installing QGroundControl and opening it for the **FIRST TIME WITHOUT ANY SETUP** of QGroundControl or setup of any pixracer board parameters/settings, it will automatically connect to the "pixracer" network connection assuming the following preconditions?
1.  OS is connected to a Pixracer network connection with SSID of "pixracer"
2. Pixracer board has latest PX4 firmware.
3. Pixracer board hardware is good.

Is the above a correct understanding of the functionality?   No wifi setup required would tell me that the connection SSID must be the default of the wifi board, never to be changed.

What about if a USB cable between computer and pixracer is installed in addition to the wifi board?   Which will QGroundControl use, usb or wifi?

I tried an experiment.  With wifi board AND usb connected to computer, (pixracer also powered by 3S battery, I launched QGroundControl.  I setup everything previously using USB and its "happy" will all the settings, compass calibrations, etc.   QGroundControl connected to the board automatically.   I then disconnected usb hoping the application would switch to wifi.  It did not and displays "waiting for vehicle connection".   I have not done any wifi settings in QGroundControl because I did not find any.
![pixracerconnectionwin7](https://user-images.githubusercontent.com/32437532/53611854-f5211100-3b9d-11e9-88eb-b63df9a5f4c0.jpg)

![image](https://user-images.githubusercontent.com/32437532/53611974-6234a680-3b9e-11e9-81ad-5de27e9a75ec.png)

**Comment by DonLakeFlyer** [comment]
> Is the above a correct understanding of the functionality?

That is correct. As long as you include 4) An anti-virus program isn't blocking QGC from seeing things on the WiFi network. QGC doesn't care/know anything about SSIDs or even WiFi really. That is all handled when you connect the OS to the PixRacer WiFi connection. There are many, many people using QGC and PixRacers out there using PixRacers with the little WiFi thingy they come with. I think there maybe an RC Groups forum which is specific to the PixRacer. You might want to ask there about people using Windows and PixRacer WiFi to see if there is some other OS level magic which may be required.

If you have an Android or iOS Phone you could install QGC and try that. That would remove Windows out of the equation which is always the most PITA OS to get things working on. If it works there, then you know there is some OS level setup problem. If it doesn't then that might point to bad hardware.

> What about if a USB cable between computer and pixracer is installed in addition to the wifi board? Which will QGroundControl use, usb or wifi?

It will connect to both and switch from one to the other as one goes away. USB connections are given priority over all other connections.

> I then disconnected usb hoping the application would switch to wifi. It did not and displays "waiting for vehicle connection".

I thought you didn't have WiFi working? It's not going to suddenly start working because you are also connected to USB.

**Comment by dogmaphobic** [comment]
Some basic answers:

The default IP address of the PixRacer WiFi is `192.168.4.1`
Once you connect your computer to it, load a browser and connect to http://192.168.4.1/getstatus
See if you get a page back with the WiFi status. If you don't, something is wrong with the ESP module on your PixRacer. If you get a status, it should be all fine.

> Does "should just work" mean that after installing QGroundControl and opening it for the **FIRST TIME WITHOUT ANY SETUP** of QGroundControl

Yes, it should automatically connect without any configuration. It's all pre configured.

> Is the above a correct understanding of the functionality? No wifi setup required would tell me that the connection SSID must be the default of the wifi board, never to be changed.

Correct. The SSID can be whatever you want. By default, it's pixracer. The SSID is only used for you, using your OSs WiFi tool, to find it and associate with it. QGC does not even know the connection is WiFi.

> What about if a USB cable between computer and pixracer is installed in addition to the wifi board? Which will QGroundControl use, usb or wifi?

They will both connect. QGC will see that both connections are from the same vehicle and assign a "priority link" to the faster link (USB in this case).

> If I'm the only person who ever had a problem with this then I'd say you guys should forget my issues and move on.

No, you are not. I've had the same issues to the point where I simply gave up playing with Windows. The issue is the OS. Not PixRacer, not QGC. When you boot the PixRacer and you are connected to its WiFi, the PixRacer will start broadcasting UDP packets. That's how QGC finds it automatically. The issue I noticed was that Windows was blocking these UDP packets and QGC never got them. I'm sure there is a way to fiddle with Windows' firewall and/or some other Windows voodoo and make that work. I just didn't have the patience to continue.

**Comment by DonLakeFlyer** [comment]
Closed this by mistake since I thought Hamish had it working. I'll take some swings through Windows connections when I get back. I have used WiFi on my Windows machine but not ESP8266. I was testing with the Sentera WiFi connection and the seems to work without problems. When you first boot QGC the OS asks for some permission on the Network which I answer yes to which I think is what opens it up.

**Comment by minerran1** [comment]
dogmaphobic,
You said "QGC does not even know the connection is WiFi."
Now I'm not a networking expert by any means but I thought the application must know which connection to use (eg. wifi, usb, etc.)  That's also why I stated to Don that I unplugged USB to see if the application would switch over to wifi (thinking there might be a software bug in QGC that this use case might overcome).    If a flight controller with wifi is also connected via usb then doesn't QGC need to choose which connection to use?   For example, I have a wireless router to the internet and I use a wireless dongle to connect to it but if I connect an ethernet cable, windows networking will (annoyingly) disconnect from wifi and start using ethernet.   So that's why I went down this path with the USB vs wifi flight controller connection.

ok I just tried dogmaphobic's experiment.  I did the following.
1. Powered up flight controller via lipo battery, no USB connection to PC.
2. Windows networking recognized and displayed the pixracer connection as available.
3. Disconnected from my wireless router connection and connected to the pixracer connection in 
the windows network and sharing center.
4. Opened a firefox tab and entered     http://192.168.4.1/getstatus
5. Browser displayed the following:

Packets Received from GCS | 0
-- | --
Packets Sent to GCS | 340
GCS Packets Lost | 0
Packets Received from Vehicle | 340
Packets Sent to Vehicle | 9
Vehicle Packets Lost | 4
Radio Messages | 0

Am I correct in concluding that the browser was able to connect to the wifi board?
I then opened QGC and it displays "waiting for vehicle connection"
Those are my findings.

**Comment by dogmaphobic** [comment]
> Now I'm not a networking expert by any means but I thought the application must know which connection to use (eg. wifi, usb, etc.)

Not really. It's a Ethernet connection. UDP to be specific. QGC doesn't know if it's using wired or WiFi Ethernet. For the application, they are all the same. QGC automatically opens an UDP port and start listening for packets received on that _known_ port (14550, which is the standard MAVLink GS port.)

Your ESP module on the PixRacer is working fine and it shows that it is transmitting. That means you have a firewall blocking it and QGC never receives these UDP packets.

**Comment by Seeelefant** [comment]
Well, that´s how the discussion started. To analyse the problem in detail you might trace the journey of your IP/UDP packets with [Wireshark](https://www.wireshark.org). Wireshark is a free and open-source packet analyzer. It is used for network troubleshooting, analysis, software and communications protocol development, and education.

**Comment by hamishwillee** [comment]
Hi @minerran1 

YOu can check what firewall software you have using
* In the Start bar, enter/select: *check firewall status*
![image](https://user-images.githubusercontent.com/5368500/53714144-6f54dc80-3ea1-11e9-9aef-2c00f4dcff4b.png)

If it is Windows Defender the instructions here should help: https://docs.qgroundcontrol.com/en/Support/Support.html#waiting_for_connection

If not, you will have to look at the specific software you are using to find out where it sets application unblocking.

**Comment by hamishwillee** [comment]
@dogmaphobic I understand how the ethernet connection works (QGC is always monitoring that UDP port, so generally it will pick up a system broadcasting).

But does it know about serial/USB connections/what it can use?
I tried testing as below but QGC connected automatically, and I still see "No serial devices available")
![image](https://user-images.githubusercontent.com/5368500/53715059-babcba00-3ea4-11e9-986f-1c5ee9c5d655.png)

I assume that both a direct USB and a telemetry radio link are both USB serial to QGC so they work in the same way?

**Comment by dogmaphobic** [comment]
> I tried testing as below but QGC connected automatically, and I still see "No serial devices available")

You won't see the "automatic" link. That's maintained by the system.



> I assume that both a direct USB and a telemetry radio link are both USB serial to QGC so they work in the same way?

Correct. They are both one and the same.

**Comment by dogmaphobic** [comment]
Baud Rate...

That brings something else to mind. Have you all set the PX4's baud rate? The one it uses to talk to the ESP8266? The screen grab below shows how I usually change it. I noticed it now says it's deprecated. I don't know what these new parameters are. @hamishwillee?

<img width="1021" alt="screen shot 2019-03-04 at 6 58 44 am" src="https://user-images.githubusercontent.com/749243/53732213-34c05500-3e4b-11e9-94f6-695582a51e7b.png">

**Comment by hamishwillee** [comment]
@dogmaphobic So in prior to PX4 1.9 the default telemetry port TELEM2 was set up using the SYS_COMPANION parameter - basically you said that the port was a companion computer, Telemetry radio, satellite radio or whatever and everything was set up for you. Pretty simple, but flawed in that it only allowed that port to be configured and there was a lot of hackering in board scripts for setting everything else. 

So in PX4 1.9 you instead can configure each service (module) to say what port it will run on - e.g. GPS service param, you specify it runs on GPS port, and configure the baud rate appropriately. This allows you to use params to configure the purpose of any port. It is a bit of a pain from a user point of view of deconflict ports, so there is a PR in QGC to switch this around for users - ie you will pick a port and get offered the service that runs on it.

- Docs on the new way of setting things here: https://docs.px4.io/en/peripherals/serial_configuration.html
- Not quite so relevant (applies to companions) here: https://docs.px4.io/en/peripherals/mavlink_peripherals.html
- Docs on this Wifi module here: https://docs.px4.io/en/telemetry/esp8266_wifi_module.html - probably should link to https://docs.px4.io/en/peripherals/serial_configuration.html too?

**Comment by minerran1** [comment]
> 
> 
> Baud Rate...
> 
> That brings something else to mind. Have you all set the PX4's baud rate? The one it uses to talk to the ESP8266? The screen grab below shows how I usually change it. I noticed it now says it's deprecated. I don't know what these new parameters are. @hamishwillee?
> 
> <img alt="screen shot 2019-03-04 at 6 58 44 am" width="1021" src="https://user-images.githubusercontent.com/749243/53732213-34c05500-3e4b-11e9-94f6-695582a51e7b.png">

Hi Hamish,
To answer your questions...
1. With regards to windows defender, yes I added QGC to the permitted apps list per someone else's suggestion on this thread even though I am not convinced that windows blocks wifi connections from applications installed locally.
2. Baud rate.   No I did not set any baud rates.  These are not applicable to USB, only to UARTs.  I know its a bit confusing because a virtual serial port is used to usb.  Anyway I've never had to set that to make a USB connection work.   I don't know much about low level windows networking details including wifi, so I can't comment on setting a data rate (baud rate) for wifi.
3.  No I did not set the sys_companion parameter.  Had no idea it existed and would never have realized that the name sys_companion = wifi.  That would be something that should be in the setup instructions (per Don's question to me earlier) if its required.  I looked thru the parameters and many of them were like "black boxes" in that I have no idea what behavior they affect.

Happy to help you guys in any way I can.   Once I understand this firmware better and have a uav flying with it, my intention is to help out as a developer.   I'm planning to git clone the repo this weekend and see if I find comments in the code that answers some of my questions.  Then at some point I can help out.

**Comment by dogmaphobic** [comment]
> So in PX4 1.9 you instead can configure each service (module) to say what port it will run on

That actually makes a whole lot more sense than those "pre-baked" parameters.

> 1. am not convinced that windows blocks wifi connections from applications installed locally.

No, it blocks connections coming _into_ applications. It's the PX4 (through the ESP8266 WiFi) that "connects" to QGC and not the other way around. I put the quotes because there is no true "connection" when it comes to UDP.

> 2. Baud rate.   No I did not set any baud rates.  These are not applicable to USB, only to UARTs.

That's the baud rate used by the Pixhawk to connect to the ESP8266 through an UART. This should be documented somewhere if it isn't, as nothing will really work if the Pixhawk is using a different baud rate than the one the ESP8266 is set to. There is a reference [here](https://docs.px4.io/en/telemetry/esp8266_wifi_module.html):

### The ESP8266 firmware has these factory settings:

- SSID: PixRacer
- Password: pixracer
- WiFi Channel: 11
- UART speed 921600

But indeed, I see no explanation about the need to set the firmware (using the parameter above). That documentation assumes the firmware installed on the PixRacer is already set when you received it.

> 3. No I did not set the sys_companion parameter.  Had no idea it existed and would never have realized that the name sys_companion = wifi.

Indeed.

In summary:

PixRacer -> UART -> ESP8266 -> UDP over WiFi -> QGC

**Comment by minerran1** [comment]
dogmaphobic,  
Thanks for the detailed explanation.

**Comment by minerran1** [comment]
Windows defender is NOT my issue by the way.  I completely disabled it and still "waiting for connection" is the result.

**Comment by hamishwillee** [comment]
@minerran1 That sux, but I do not know what further advice to give you. Hopefully Don or Gus will have other ideas. 

In the meantime, starting to capture some of the things we know are issues:
1. The ESP8266 setup instructions have been updated to reflect the baud rate changes + tidy up some other inconsistencies: https://github.com/PX4/px4_user_guide/pull/459

**Comment by minerran1** [comment]
Perhaps the default baud rate of the flight controller used to talk to the ESP8266 could be the same as the default rate the ESP8266 firmware uses?  It must not be otherwise dogmaphobic would not have said to check it.   Better still, if the flight controller used the autobaud algorithm to discover the rate.

**Comment by hamishwillee** [comment]
@minerran1 Whether using SYS_COMPANION or the new new serial port approach you have to configure the baud rate for the port, because the mavlink module has no idea what hardware it is talking to. If you hard code a port baud rate for ESP8266 every other will no longer be optimised.

Auto baud rate detection might be convenient though. @bkueng can/do we support this in a generic way?

**Comment by minerran1** [comment]
> @minerran1 Whether using SYS_COMPANION or the new new serial port approach you have to configure the baud rate for the port, because the mavlink module has no idea what hardware it is talking to. If you hard code a port baud rate for ESP8266 every other will no longer be optimised.
> 
> Auto baud rate detection might be convenient though. @bkueng can/do we support this in a generic way?

Good point.   It cannot be hard-coded.   So I would just suggest that in the user's manual, it be explained clearly (so that rookies will understand), that the parameter SYS_COMPANION baud rate must be changed to <value for ESP8266>.   Having a section detailing how to find and change parameters might not be a bad idea also.  I had a general idea, having come from using INAV open source FC configurator.

**Comment by minerran1** [comment]
Guys, not sure what happened but **its now working**.  I did a PX4 firmware upgrade on the flight controller and set the SYS_COMPANION baud rate to 921600.  The day I made the change, it didn't work.   Today, starting fresh, it does.  Maybe I needed to power cycle after changing the baud rate.  Or maybe it was the firmware upgrade?  Who knows.  You can see the wifi-bridge displayed in the screen cap below.  Perhaps that pic could be included in the manual to show what is seen when QGC connects to the wifi?
![GGroundControlWifi](https://user-images.githubusercontent.com/32437532/54721139-91a05880-4b37-11e9-932f-802d61c7c6b4.png)

![WinNtwkPropertiesPixracer2](https://user-images.githubusercontent.com/32437532/54721696-032cd680-4b39-11e9-9005-f846625aea20.png)
![WinNtwkPropertiesPixracer1](https://user-images.githubusercontent.com/32437532/54721705-0922b780-4b39-11e9-9778-d0a416734abe.png)

![PixracerConnection](https://user-images.githubusercontent.com/32437532/54721908-a54cbe80-4b39-11e9-93bf-4efbcc051600.png)

**Comment by hamishwillee** [comment]
@minerran1 Great news !!!!
1. Yes a reboot is required after setting [SYS_COMPANION](https://docs.px4.io/en/advanced_config/parameter_reference.html#SYS_COMPANION). QGC prompts you of this, but sometimes this is not obvious. YOu can also see what params require reboot in the docs (it says "Reboot required: true" at the end of each param doc.
1. Yes, I will show what a successful connection looks like in the ESP doc update - https://github.com/PX4/px4_user_guide/pull/459
1. 
   > Having a section detailing how to find and change parameters might not be a bad idea also. I had a general idea, having come from using INAV open source FC configurator.

   The [doc here](https://docs.px4.io/en/advanced_config/parameters.html) shows how to set parameters. I had not linked this from the ESPxx setup instructions because the instructions are linked in the "set the serial port configuration" doc where you use them. I have however now added it as part of a note for using SYS_COMPANION.
1. Are you otherwise happy for this issue to be closed?

**Comment by hamishwillee** [comment]
PS, @minerran1  https://github.com/PX4/px4_user_guide/pull/459 updates the ESP doc using your images.

**Comment by minerran1** [comment]
I agree that this issue should be closed.  thanks!

**Comment by hamishwillee** [comment]
@DonLakeFlyer As discussed above, I believe this can be closed (all users in discussion have working setup, and docs capture the workarounds/issues - though still waiting on Gus to review https://github.com/PX4/px4_user_guide/pull/459)
